### PR TITLE
Provide an option to use netty's round robin inet address selection

### PR DESCRIPTION
**Pull request by ybayk**
Signed-off-by: ybayk yurgis.baykshtis@clarivate.com

### CODE REVIEW

**Comment by yurgisbaykshtis** [comment]
Signed-off-by: ybayk yurgis.baykshtis@clarivate.com

File: src/test/java/io/vertx/core/dns/HostnameResolutionTest.java [hunk 1]
```diff
@@ -834,6 +840,74 @@ private void testServerSelection(boolean rotateServers, boolean cache) throws Ex
...
+
+      @Override
+      public String get(String s) {
+        return DnsAttribute.IP_ADDRESS.equals(s)? ipAddress : null;
+      }
+    };
+  }
+
+  @Test
+  public void testRoundRobinInetAddressSelection() throws Exception {
```
Comment (vietj) [inline src/test/java/io/vertx/core/dns/HostnameResolutionTest.java]: I think we are missing a test where we check that setting setRoundRobinInetAddress to false only returns the first one.

Comment (yurgisbaykshtis) [reply src/test/java/io/vertx/core/dns/HostnameResolutionTest.java]: addressed, now we have 3 tests - default, on, and off.

Comment (vietj) [reply src/test/java/io/vertx/core/dns/HostnameResolutionTest.java]: nice

**Comment by vietj** [comment]
can you sign the Eclipse Contributor Agreement and provide a new PR with commits signed ?

**Comment by yurgisbaykshtis** [comment]
i did sign eca (but after the i created pr). any of my attempt to satisfy the check failed. do i have to create a brand new pr? also how and where exactly do i sign? in the commit message  or during pr creation? thanks!

**Comment by vietj** [comment]
can you read this https://github.com/eclipse-vertx/vert.x/blob/master/CONTRIBUTING.md and submit a new clean PR ?

**Comment by bolbat** [comment]
Hello
Any news about this improvement? 
We got problem with some DNS servers who return IP's in static order without rotation (for example google do it in this way (8.8.8.8)) and this improvement will resolve it. 
@vietj Would be nice to have it in 3.9.2

**Comment by vietj** [comment]
@ybayk can you provide a new PR with Eclipse Agreement so we can progress on this ?

**Comment by vietj** [comment]
@ybayk can you provide a clean PR and sign the contributor agreement ?

**Comment by vietj** [comment]
I can still see that the ECA check is not passing

**Comment by tsegismont** [comment]
Closing, superseded by #3824

**Pull request closed without merging**